ginga/misc/plugins/Thumbs.py

```diff
@@ -331,11 +337,13 @@ def focus_cb(self, viewer, fitsimage):
             # No memory of this thumbnail, so regenerate it
             chname = viewer.get_channelName(fitsimage)
             self._add_image(viewer, chname, image)
+            self.highlight_thumbnail(viewer, fitsimage)
             return
 
         # Else schedule an update of the thumbnail for changes to
         # cut levels, etc.
         self.redo_delay(fitsimage)
```

Comment:
```python
if self.highlight_tracks_keyboard_focus:
    self.highlight_thumbnail(viewer, fitsimage)
```

This is a callback for keyboard focus.  So if you are wanting highlight based on that, we check preference and if set, highlight.